google_guest_agent/addresses.go

```diff
@@ -503,7 +503,7 @@ func enableNetworkInterfaces() error {
 	switch {
 	case osRelease.os == "sles":
 		return enableSLESInterfaces(googleInterfaces)
-	case (osRelease.os == "rhel" || osRelease.os == "centos") && osRelease.version.major >= 7:
+	case isNetworkManagerRunning():
```

Comment:
you're mixing different sorts of tests into the case. what if there was a sles system running network manager?

additionally, i think we only want to create this file on EL-type distros that will respect the /etc/sysconfig system that we modify with the `disableNM` function. if NetworkManager is running on Debian, I am not sure anything would need to be done. Will look into that.